### ISSUE TITLE
Looking for an option to fix tab title for Version 3.1.x

### ISSUE THREAD

**Issue by llinfeng** [jupyterlab/jupyterlab#10820]
### Problem
It is distracting and impractical to have the following ever-changing tab-title when I switch across multiple notebooks in a workspace.
![CannotFix_TabTitle](https://user-images.githubusercontent.com/5890407/129217511-f2761865-1547-4b05-8b6b-4bb9d7b83f40.gif)

The old way of fixing the tab title no longer works as intended. Upon fixing the tab title in one notebook, switching to another notebook resets the tab title. Even switching back to the notebook that ran the following old script won't restore the tab name as "A fixed title".
```
%%javascript
document.title='A fixed title'
```

### Proposed Solution

I may have missed something well documented. If so, please point me there. If not, it helps to know what other `%%javascript` snippets we can use to get the tab title fixed.

### Additional context

I am current running upyterlab Version 3.1.4. I prefer to have a fixed tab title since I rely on it to identify the "type of window". This helps to issue additional AutoHotKey configurations to, say, prevent the Print interface from popping up with `Ctrl+p`. I also use it to map `Ctrl+]` as jump back to the cell-selection layer.

**Comment by krassowski** [comment]
I agree that this should be user-customizable. This was originally suggested in https://github.com/jupyterlab/jupyterlab/issues/6946#issuecomment-664604505 and some suggestions on variables to be made available is in https://github.com/jupyterlab/jupyterlab/issues/6946#issuecomment-675804557.

**Comment by krassowski** [comment]
>  I prefer to have a fixed tab title since I rely on it to identify the "type of window". This helps to issue additional AutoHotKey configurations to, say, prevent the Print interface from popping up with Ctrl+p. I also use it to map Ctrl+] as jump back to the cell-selection layer.

https://xkcd.com/1172/ ;) [Just joking, I fully appreciate your use case and as mentioned earlier I do agree this should be improved].

In case if you would like to take a stab at it, the PR which introduced this feature is: https://github.com/jupyterlab/jupyterlab/pull/10002

**Comment by llinfeng** [comment]
@krassowski Thanks a lot for pulling the relevant threads together. As advised by <https://xkcd.com/1205/> (seriously, lol), I will live with my current workaround based on the following observation:
* When the tab-title updates, it always contains the first 15 characters of the current working directory for the noetbook.

Since I don't have too many variants for the working directories, I will add a few more lines to my AutoHotKey script to take note of these directory names. This will break, well, should the name of the notebook gets the spotlight in the tab-title area :) Glad this is not happening with Version 3.1.4.

PR #10002 looks really promising. I am not confidence if I can get it up and running locally in a reasonably short period of time. :)
